crypto/property/property_parse.c

```diff
@@ -567,8 +567,12 @@ OSSL_PROPERTY_LIST *ossl_property_merge(const OSSL_PROPERTY_LIST *a,
         r->has_optional |= copy->optional;
     }
     r->num_properties = n;
-    if (n != t)
-        r = OPENSSL_realloc(r, sizeof(*r) + (n - 1) * sizeof(r->properties[0]));
+    if (n != t) {
```

Comment:
I agree, but I was uncertain if there was any dependency of this result? Since you wrote the original code I feel safe to delete the realloc.
Semi-related question: why do we only allocate room for t-1 elements in the resulting list? Shouldn't
```
    const int t = a->num_properties + b->num_properties;
    r = OPENSSL_malloc(sizeof(*r)
                       + (t == 0 ? 0 : t - 1) * sizeof(r->properties[0]));
```
be 
```
    const int t = a->num_properties + b->num_properties;
    r = OPENSSL_malloc(sizeof(*r) + t * sizeof(r->properties[0]));
```
?